### ISSUE TITLE
PostgreSQL schema

### ISSUE THREAD

**Issue by kesavkolla** [go-xorm/xorm#361]
I've all my tables in a separate schema.  I don't use **public** schema at all.  How can I set schema name at DB level so that all my types work seamlessly.

**Comment by lunny** [comment]
I think this a bug of xorm. I will resolve this issue ASAP. @kesavkolla #286

**Comment by lunny** [comment]
I have push a comimt to resolve this issue. Please help to test it. @kesavkolla @tossp
You should add schema parameter on connection string if you have a specify schema not named 'public', for example,
```Go
xorm.NewEngine("postgres", "dbname=xorm_test sslmode=disable schema=public")
```

**Comment by tossp** [comment]
in`Information.go:236`
`models.DB.Where("information_order_uid = ?", m.Uid).Get(&obj)`
```
2016/03/08 02:11:25 [router.go:854][C] Handler crashed with error reflect: NumField of non-struct type
2016/03/08 02:11:25 [router.go:860][C] D:/Develop/Go/src/runtime/asm_amd64.s:472
2016/03/08 02:11:25 [router.go:860][C] D:/Develop/Go/src/runtime/panic.go:426
2016/03/08 02:11:25 [router.go:860][C] D:/Develop/Go/src/reflect/type.go:660
2016/03/08 02:11:25 [router.go:860][C] D:/src/golang/src/github.com/go-xorm/xorm/engine.go:926
2016/03/08 02:11:25 [router.go:860][C] D:/src/golang/src/github.com/go-xorm/xorm/engine.go:854
2016/03/08 02:11:25 [router.go:860][C] D:/src/golang/src/github.com/go-xorm/xorm/engine.go:876
2016/03/08 02:11:25 [router.go:860][C] D:/src/golang/src/github.com/go-xorm/xorm/session.go:1035
2016/03/08 02:11:25 [router.go:860][C] D:/src/golang/src/its/models/quality/Information.go:236
```

**Comment by tossp** [comment]
Sync2同步关系的时候，貌似没有生效

`尝试插入菜单 &pq.Error{Severity:"ERROR", Code:"42P01", Message:"relation \"zh.menu\" does not exist", Detail:"", Hint:"", Position:"22", InternalPosition:"", InternalQuery:"", Where:"", Schema:"", Table:"", Column:"", DataTypeName:""
, Constraint:"", File:"src\\backend\\parser\\parse_relation.c", Line:"965", Routine:"parserOpenTable"}`

**Comment by lunny** [comment]
OK. 我看下。@tossp

**Comment by lunny** [comment]
I think maybe #379 is a better solution for schema of pg, so that I will close this issue and revert the old method. That means it will not be supported schema parameter on connection string. @tossp

**Comment by lunny** [comment]
closed by #876
